src/library_fetchfs.js

```diff
@@ -5,10 +5,19 @@
  */
 
 addToLibrary({
-  $FETCHFS__deps: ['$stringToUTF8OnStack', 'wasmfs_create_fetch_backend'],
+  $FETCHFS__deps: ['$stringToUTF8OnStack', 'wasmfs_create_fetch_backend', 'wasmfs_fetch_create_manifest', 'wasmfs_fetch_add_to_manifest'],
   $FETCHFS: {
     createBackend(opts) {
-      return _wasmfs_create_fetch_backend(stringToUTF8OnStack(opts.base_url));
+      var manifest = 0;
+      if (opts['manifest']) {
+        manifest = wasmfs_fetch_create_manifest();
+        Object.entries(opts['manifest']).forEach((pair) => {
+          var path = stringToUTF8OnStack(pair[0]);
+          var url = stringToUTF8OnStack(pair[1]);
+          wasmfs_fetch_add_to_manifest(path, url);
+        })
+      }
+      return _wasmfs_create_fetch_backend(stringToUTF8OnStack(opts.base_url), opts.chunkSize | 0, manifest);
```

Comment:
If you use `stringToUTF8OnStack` in a JS function you need to somehow restore the SP at the end of the function.   One simple way to do that is wrap the whole thing in `withStackSave(() => ...)`

Reply:
OK--the first argument to create_fetch_backend was like this before, and did not use withStackSave.  Should I wrap that call in the withStackSave function as well?

Reply:
Ah yes, that looks like an existing bug (a leak of stack space).

Reply:
Done.